### DISCUSSION TITLE
Ongoing design work for Jellyfin ATV

### DISCUSSION THREAD

**Discussion by MS-design-ui** [jellyfin/jellyfin-androidtv#807] [General]
So as mentioned and shown in Element, I'd love to help out with designing a perfect experience for Jellyfin.

### Background

I'm a big fan of jellyfin, but I find certain things difficult to use in the current state. I'd like to help on an ongoing basis with UI and the UX of things.

### Log-in (WIP)

![Login - Less focus on server](https://user-images.githubusercontent.com/49571600/113894504-f92dad00-97bf-11eb-99f1-79d1678efae0.png)

The new design allows the feel for multiple users, and the ease of multiple servers. Seeing as most users (Assumption) have one primary server, they'd really be connecting to only one server at any time. The idea relies on recently logged in, followed by alphabetical results (fed from the server if the user is broadcasted)

### Home screen
![Main Menu](https://user-images.githubusercontent.com/49571600/113893789-3fced780-97bf-11eb-95a2-98e9003fee5b.png)
![Main Menu_Expanded](https://user-images.githubusercontent.com/49571600/113893800-41989b00-97bf-11eb-8c65-fd7c965065e6.png)

The new home screen allows a resemblance with the web UI and other major streaming platforms. It's following branding from the JF branding site. Where I could use creative freedom, I have done.

### Detail page view

![Details-Blue](https://user-images.githubusercontent.com/49571600/113894067-8f150800-97bf-11eb-9a24-3c0f35f31025.png)

I've done this for movies only, for now, the TV shows will require a little more thought since there are multiple things within that.

### Style Guide
**Typography**
![Typography + Modals](https://user-images.githubusercontent.com/49571600/113894880-5590cc80-97c0-11eb-9b80-84d825baf6bd.png)

**Icons**
![Iconography](https://user-images.githubusercontent.com/49571600/113894882-56296300-97c0-11eb-89e9-168502d66976.png)

**Shared components**
![Re-used elements](https://user-images.githubusercontent.com/49571600/113894887-56296300-97c0-11eb-9014-734cd3351705.png)

**Buttons**
![Buttons](https://user-images.githubusercontent.com/49571600/113894891-56c1f980-97c0-11eb-9fda-b8247a74d399.png)

**Colours**
![Colours](https://user-images.githubusercontent.com/49571600/113922132-d19a0d00-97de-11eb-908d-3e067b44f610.png)

I'll update progress in this discussion, and would love feedback on designs. After all, this is a community-driven project :)

**Comment by MS-design-ui** [comment]
Had some time to work on the TV view. Here's how it's looking

![Details (TV)](https://user-images.githubusercontent.com/49571600/113927809-dca46b80-97e5-11eb-8bee-bbafaa5d7858.png)

![Details (TV Episode selected)](https://user-images.githubusercontent.com/49571600/113927807-db733e80-97e5-11eb-9a2b-d271c0a8d027.png)

**Reply by dannycool87** [reply]
WoW, I want this asop in the app!!!
Very good job!

**Reply by Sledgehamma** [reply]
Looks absolutely gorgeous and very clean! Just my two cents:

- When in the details pages of a movie or TV show the left menu should be hidden
- I’d like some logos about the quality of that movie/TV show like Dolby a vision, HDR10, 4K, etc.

**Comment by thornbill** [comment]
Should the home screens use the subtle gradient? I was thinking maybe the side panel would be solid colored and the overall background could have the gradient.

**Reply by MS-design-ui** [reply]
![Main Menu - Expanded_Gradient](https://user-images.githubusercontent.com/49571600/114189197-d5927000-9941-11eb-84b5-8e0c38b03e67.png)
Like this?

**Reply by thornbill** [reply]
Yeah that's what I was thinking.

**Reply by MS-design-ui** [reply]
I'll check the accessibility. If that's okay then we're good to go :)

Main reason why I didn't want the gradient there was for focus. The bright colours of the posters and font should really be the main focal attention imo.

**Reply by thornbill** [reply]
Yeah fair enough and I'm not a designer, so take my recommendations with a grain of salt lol

**Comment by neilsb** [comment]
Firstly, really like how this is looking.  Just a couple of thoughts:

A general thought is on the sizing of items with respect to a TV interface.  I've not tried it on a  TV, but I'm guessing things would be quite small looking, and potentially not easy to read from a couch a few meters away.   Not tried it as I say, so could be very wrong, but just comparing to what other TV interfaces look like it seems like it may need making a bit larger.

Secondly is the SyncPlay button at the top of the screen.   If you're scrolling down through a library, then getting back up to the top of the screen for _any_ controls can result in a lot of scrolling work.   I understand it's probably there as that's where it is in web, but I think it may make more sense to have it in the menu bar at the side, just making it easier to get to in most circumstances with fewer button presses.

**Reply by MS-design-ui** [reply]
Thanks for the contribution. The resolution is defo something I need to test. I'm designing on a desktop and I do not know how it will translate to a TV....

The size of the elements can easily be increased however :)

RE: Syncplay
I'm struggling to justify putting it in the side bar. I'm not sure how often it will be used to be completely honest with you. If you/anyone reading this would like to comment on your usage that would be extremely helpful.
It can still be gotten to pretty quick as well, but it would mean navigation to another page so you're back at the top...

Awaiting more user insights for the above :)

Thank you for your involvement!

**Reply by MS-design-ui** [reply]
![Main Menu - Expanded](https://user-images.githubusercontent.com/49571600/114305310-909c4400-9acf-11eb-8d42-13345ad1b024.png)
![Uploading Main Menu.png…]()

Increased the sizing of pretty much all the elements. I shared the same fear as you :)

**Reply by neilsb** [reply]
I'm thinking we could also look at using this for the Roku Client as well, if there are no objections from yourself or the ATV team.

**Reply by MS-design-ui** [reply]
> I'm thinking we could also look at using this for the Roku Client as well, if there are no objections from yourself or the ATV team.

Go wild :)

Give me some time to fully finalise the components though. I don't want you / others to re-work :P

**Reply by heyhippari** [reply]
Regarding sizing, the Android TV documentation has some guidelines: https://tv.withgoogle.com/style/typography.html

**Reply by dkanada** [reply]
I'm also not keen on the SyncPlay button. I don't mind the placement but the style looks out of place to me. The items next to it look so sleek in comparison, the button stands out with that heavy background.

**Reply by MS-design-ui** [reply]
@MrTimscampi Thank you!

I'll check it against my designs atm. I'm pretty sure I meet the reqs though :)

**Comment by MS-design-ui** [comment]
Worked on this a little more today

Changes:
* Made the designs consistent across the screens
* Created a Library view (for movies)
* Watched states and multiple files counters work together (For the movie view)
* Accessibility tested
* Spacing / Fonts increase to work on smaller screens
* Mouse-Over to show where cursor is

![Main Menu](https://user-images.githubusercontent.com/49571600/114307377-4323d500-9ad7-11eb-8d84-2f7b7b87f969.png)

![Main Menu - Expanded](https://user-images.githubusercontent.com/49571600/114307368-3b643080-9ad7-11eb-94b0-076cd2cf9aef.png)

![Library](https://user-images.githubusercontent.com/49571600/114307358-2d161480-9ad7-11eb-88e7-fefb9b700f4a.png)

**Reply by MS-design-ui** [reply]
When multiple filters are applied, or even one, these are the views:
![Library - Filter(s) Applied](https://user-images.githubusercontent.com/49571600/114308005-305ecf80-9ada-11eb-8000-15980e300916.png)


![Library - Multiple Filter(s) applied](https://user-images.githubusercontent.com/49571600/114307991-263cd100-9ada-11eb-80b4-a4e4126aa7c7.png)

When a filter is applied, the remove filter button also becomes available for ease of UX

**Comment by Bitwolfies** [comment]
Absolutely amazing, does any part of it have code behind it, or is it just a concept? Cant wait for androidtv to look like this.

**Reply by nielsvanvelzen** [reply]
It's all concept art right now. We'd love to make these designs a reality but that will take quite some time, I'm prioritizing the stability of the app (like rewriting the playback code) above redesigning right now.

**Reply by Bitwolfies** [reply]
As you should, cant wait to see it.

**Reply by finnzz** [reply]
This may be an unpopular opinion, but I find myself avoiding regular use of the AndroidTV app because of the layout / look over any stability issues.  I'm eager to see this much improved look, as well as enabling skins in the app.  Don't underestimate the value of pleasant looking :)

**Reply by jedicoach44** [reply]
> This may be an unpopular opinion, but I find myself avoiding regular use of the AndroidTV app because of the layout / look over any stability issues. I'm eager to see this much improved look, as well as enabling skins in the app. Don't underestimate the value of pleasant looking :)

You’re 100% correct. The average end user…. Not the person setting up the server and going through all that… but the family me never who wants to watch a movie or show…. They want it to look beautiful and work flawlessly. To many apps out there to have an “out dated” app as your primary way of viewing things.

**Reply by nielsvanvelzen** [reply]
So you'd rather have an app that looks pretty but crashes when you try to play a video instead of an app that doesn't look super pretty but is able to play the video?
Don't get me wrong. I, too,  wouldn't to use an app that looks ugly. But with the upcoming 0.12 release the app already looks way modern enough in my opinion. It even supports themes :)

**Comment by dkanada** [comment]
Have you considered what the cards should look like when images aren't available? Lots of content might not be matched correctly, and I'm sure many users won't be setting their avatars. Absolutely loving the designs so far though! Do you mind if I use any of these ideas in Gelli for Android?

**Reply by MS-design-ui** [reply]
I haven't. I am not sure if this is something that the apps focus on, that'd be more the server. Correct me if I am wrong @thornbill 

Also yes, you can use it for Gelli, but I'm not sure it will help you from a UX perspective. I've mainly focussed for video content :(

Feel free to take inspiration from it though 👍

**Reply by thornbill** [reply]
It's definitely possible for media to be missing images. Right now we just show a solid color with an icon depending on the type of media that it is as a placeholder in that case. Ideally most media should have images though.

**Comment by MrChip53** [comment]
Can you do a TV guide mock up for me? :D

**Reply by nielsvanvelzen** [reply]
Oh you want to work on that? 👀

**Reply by MrChip53** [reply]
Yes! I already started a long time ago but I'm no designer so I stopped when I had to redesign everything lol

**Reply by MS-design-ui** [reply]
Have you got anything so far? Just so I know functionality?

**Reply by MrChip53** [reply]
Elaborate what you mean please? I pretty much want a design that looks like a modern cable box guide. Channel headers on left programs to the right. No pagination. A settings button will need to be somewhere.

**Reply by MS-design-ui** [reply]
@MrChip53 I do not have a Live TV sub in any of my media servers, so I'm not entirely familiar with the use cases it provides. What I'd love is to see if there is something already in Jellyfin, primarily to extract the functions it has. I'll re-work the UI and UX with the existing set of functions if possible, else I'll add a few functions which may require more time.

**Reply by anthonylavado** [reply]
@RemuxKid There is some stuff in the main web app already, Roku, and of course Android TV. I'll get a set of screenshots from them for you.

**Reply by MS-design-ui** [reply]
@anthonylavado Thanks :)

Atm, my JF is buggy because of library size, so not being able to utilise it fully :(.

Mainly waiting for the DB re-write :D

**Reply by MrChip53** [reply]
Sorry I completely forgot about this but currently for the rewrite all I have is a very very rough design that would never make it to final.

Here are some Roku screens though https://github.com/jellyfin/jellyfin-roku/issues/336

**Comment by jedicoach44** [comment]
These are all awesome. I’m a graphics designer at heart and would love to help with these and some others for Jellyfin. My expertise is in visual consistency, proper branding, usability and more…. All on cross platform solutions. If you’re open to help, let me know!

**Reply by Bitwolfies** [reply]
Yknow I remember one of the devs saying jf lacks a design language, maybe see about that? You should join the matrix/discord/irc (it’s all the same chat you can just join how you like :))

**Reply by MS-design-ui** [reply]
@jedicoach44 If you're comfy with UI, I can invite you to the Figma file I'm editing?

**Reply by jedicoach44** [reply]
Absolutely. I'd love to get in there and take a look and help in any way I can!

**Comment by dannycool87** [comment]
It is ongoing in 0.12 to have the possibility to choose the video version of a title? Like between 4k and 1080p like the audios and subtitles chooses? It's a big issue for me... Thanks!

**Reply by anthonylavado** [reply]
This is best as a separate thread/issue (one likely exists), rather than in this discussion about design. Nothing personal, just trying to stay focused 🙂

**Reply by dannycool87** [reply]
No problem, I understand. I speak of it there, because this option have to be integrated to the design and thinking of the best way to it. I say it, because the way to choose the audio and subtitles actually doesn't look pretty good, probably to short to show correctly the text in. Thank you for your work!

**Reply by nielsvanvelzen** [reply]
FYI: we can add missing features without rewriting the complete UI. And even if a feature is seen in a mockup, it won't mean it will be implemented when creating the code for that UI.

**Comment by MrMoreira** [comment]
I'm a mere jellyfin user, I would like to know if someone has already started to develop if they have a beta app or is it just a concept?

**Reply by nielsvanvelzen** [reply]
All concept

**Comment by ClappTV** [comment]
There are screens that look somewhat empty in the designs, perhaps putting a background image at the top right would give more visibility to the screens.

something like that:

![image](https://user-images.githubusercontent.com/6973905/126865603-8ddaab64-4141-45dd-a08b-b19770369f9a.png)
![image](https://user-images.githubusercontent.com/6973905/126865607-495e483b-c9f3-4aaf-8492-fca4d93e418d.png)

It seems to me that the interface is beautiful and more dynamic

**Comment by ggowins** [comment]
For LiveTV, would it be possible to have an option for the Guide to not have pagination, and just have all of the channels in a single list, like the web client or Jellyfin Media Player app?

**Reply by MrChip53** [reply]
That will be part of the goal with the new guide, yes.

**Comment by joebckFeg8** [comment]
This looks absolutely incredible. Great job !

**Comment by FancyNerd92** [comment]
Try to come closer with emby android tv app, it's more like Netflix style. For example use the style of Jellyfin Media Player for Windows, it's really awasome and you can view your theme on this and your custom logo. It will be perfect if we can add our logo on Apps (Android TV, Samsung TV etc) as iOS App and Jellyfin Media Player. I'm web designer too, if you want anything tell me 😉

**Comment by joebckFeg8** [comment]
I've had a look around for an update on the GUI for Android tv but can't seem to get a concise answer of what the current status is or whether it's even begun to move from concept visuals only. Does anyone know? 🤷‍♀️

**Reply by nielsvanvelzen** [reply]
The latest version (0.12) contains changes to the user interface. There hasn't been much visual changes since. This discussion contains concept art only.

**Comment by fmchris10** [comment]
I like the mocked up designs, the blue looks super cool, but I hate menu's on the side. I know Jellyfin is based on Plex, but I would hope things lead away from their style (which is one of my main gripes).

The current design could easily be spruced up and, improved by a designer.

Elegent and simplicty is the key.

**Reply by thornbill** [reply]
Jellyfin started as a fork of Emby 3 years ago when Emby went closed source. (The Android TV app had been closed for many years prior to that though.) There is no direct relation to Plex. 🙂

**Comment by tangsgod** [comment]
It looks really great !
One year after, is it still a concept or did you managed to code the dream ?  😃

**Comment by tangsgod** [comment]
Or even a little update on where the project is ?

**Comment by teobaranga** [comment]
Any updates on this? Would be possible to share the Figma project so that other people could pick it up and contribute? It's a really great design 🔥

**Comment by ghost** [comment]
Any update on this? Would be awesome to see this done.

**Comment by KakarottoCake** [comment]
An update on this would be amazing

**Reply by thornbill** [reply]
There isn't really anything to update. I don't think the person who started working on this design has been active in a long time.

**Reply by KakarottoCake** [reply]
That's really a shame. I hope the source code at least is released so someone else can continue it...

**Reply by nielsvanvelzen** [reply]
There is no source. It's just a few mockups.

**Reply by finnzz** [reply]
Reminds me of this Reddit Jellyfin post that got a lot of attention, but never materialized

https://www.reddit.com/r/jellyfin/comments/10buwhw/jellyfin_react_web_client_looking_for_prealpha/

**Comment by trickbaze** [comment]
Is there a ETA for releasing this?

**Comment by mels0n** [comment]
I'm wondering if this is the path forward, or is there a path forward in which the Android TV app merely uses the Web UI as the face, similar to how the android app works, or something else entirely?

**Comment by TacoTakumi** [comment]
Hey all - jumping in to offer help building this, not just admiring it.

About me: I've been doing software for 30+ years and was a core dev on Freevo (the old Python media-center/PVR project), so TV UIs are home turf. These days I'm a daily Jellyfin user on a Shield.

@MS-design-ui's mockups still look great and the "is this still just a concept?" question that keeps coming up here nails the problem: the design isn't the blocker, the implementation is, and that's tied to the Leanback to Compose migration.

I've been poking around the codebase and think I get the lay of the land migration's going foundation-first (Compose host, router/backstack, interaction tracking, LegacyImageCardView redone in Compose), and it's plain Compose on your own ui/base components + JellyfinTheme, not androidx.tv. I threw together a little Compose "modern home" prototype (hero + landscape rows) on top of that and ran it on my Shield to feel it out.

Mostly I just want to help however actually fits your plan, not lob a redesign over the wall. A few questions:

- Is there a rough roadmap / order for migrating screens, and where do you coordinate it here or Element?
- Anything small I could pick up as a first contribution, in your existing style?
- Down the line, any interest in pulling shared styling into design tokens in :design (card shapes, aspect ratios, spacing, hero) so the look's tunable in one place?
- PR #5146 (Compose BrowseGrid) looks close to what folks keep asking for. Is it worth helping push over the line, or not the direction you want?

Happy to start small and follow your lead. Thanks for all the work.

-TT
